scanner/src/main/kotlin/Scanner.kt

```diff
@@ -356,9 +356,14 @@ class Scanner(
                     "Starting scan of '${referencePackage.id.toCoordinates()}' with package scanner '${scanner.name}."
                 }
 
-                // Filter the scan context to hide the excludes from scanner with scan matcher.
-                val filteredContext = if (scanner.matcher == null) context else context.copy(excludes = null)
-                val scanResult = scanner.scanPackage(referencePackage, filteredContext)
+                val adjustedContext = context.copy(
+                    // Hide excludes from scanners with a scanner matcher.
+                    excludes = context.excludes.takeUnless { scanner.matcher != null },
+                    // Tell scanners also about the non-reference packages.
+                    coveredPackages = packagesWithIncompleteScanResult
+                )
+
+                val scanResult = scanner.scanPackage(referencePackage, adjustedContext)
```

Comment:
Related to my comment in line 63: We could alternatively pass the `coveredPackages` as additional parameter to `scanPackage()` or pass a dedicated `PackageScanContext` class, so that we don't need to worry in other places about this new property.  What do you think?

Reply:
> pass the `coveredPackages` as additional parameter to `scanPackage()`

As indicated in the commit message, I've not done it that way to avoid a breaking API change in the unction signature. Also, including `coveredPackages` into `ScanContext` paces the way for eventually also teach `ProvenanceScannerWrapper`s which packages are contained in the passed `provenance`.

Reply:
I think there's no general problem with having properties in the `ScanContext` that are not used by all scanners. For example, `ScanContext.labels` is only used by `FossId` and other scanners can simply ignore it.